ignite/contrib/handlers/base_logger.py

```diff
@@ -172,17 +172,17 @@ def attach(
         Returns:
             :class:`~ignite.base.events.RemovableEventHandle`, which can be used to remove the handler.
         """
+
         if isinstance(event_name, EventsList):
             for name in event_name:
-                if name not in State.event_to_attr:
+                if name not in engine._allowed_events:
```

Comment:
Do we have a public method to check that ?

Reply:
That's already tested everywhere in `tests/ignite/contrib/engine/test_common` and other places also

Reply:
In tests it is OK, but in the code it is a bit different. It is only used in 3 places: https://github.com/pytorch/ignite/search?q=_allowed_events . OK, for now to use it like that. Maybe, later we can create a public method for that.